frontend/src/components/pages/CampOverview/ManageSessions/ManageSessionsModal.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from "react";
+import {
+  Button,
+  HStack,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from "@chakra-ui/react";
+import { ManageCampSessionDetails } from "../../../../types/CampsTypes";
+import ManageSessionsModalTable from "./ManageSessionsModalTable";
+
+interface ManageSessionsModalProps {
+  campStartTime: string;
+  campEndTime: string;
+  sessions: Array<ManageCampSessionDetails>;
+  onSaveChanges: (
+    deletedSessions: Set<string>,
+    updatedCapacities: Map<string, string>,
+  ) => void;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const ManageSessionsModal = ({
+  campStartTime,
+  campEndTime,
+  sessions,
+  onSaveChanges,
+  isOpen,
+  onClose,
+}: ManageSessionsModalProps): JSX.Element => {
+  const [deletedCampSessionIds, setDeletedCampSessionIds] = useState<
+    Set<string>
+  >(new Set());
+  const [updatedSessionIdCapacity, setUpdatedSessionIdCapacity] = useState<
+    Map<string, string>
+  >(new Map());
+
+  const closeModalAndClearChanges = () => {
+    setDeletedCampSessionIds(new Set());
+    setUpdatedSessionIdCapacity(new Map());
+    onClose();
+  };
+
+  return (
+    <Modal isOpen={isOpen} onClose={closeModalAndClearChanges} isCentered>
+      <ModalOverlay />
+      <ModalContent pt={5} minW="1000px" maxH="600px">
+        <ModalHeader ml={6} textStyle="heading">
+          <Text>Manage Camp Sessions</Text>
+        </ModalHeader>
+        <ModalBody>
+          {sessions.length === 0 ? (
+            <Text>No sessions to edit</Text>
+          ) : (
+            <ManageSessionsModalTable
+              campStartTime={campStartTime}
+              campEndTime={campEndTime}
+              sessions={sessions}
+              deletedSessions={deletedCampSessionIds}
+              updatedCapacities={updatedSessionIdCapacity}
+              onEditCapacity={(updatedSession, updatedCapacity) => {
+                // TODO check for duplicate calls (user types back in original capacity)
```

Comment:
did you wanna do that in this pr?

Reply:
hmm I'll keep this out for now, there's 3 TODOs that I can think of:
- this one
- add tooltip for error state
- disable save button when invalid capacity